domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/showcondition/ShowConditionParser.java

```diff
@@ -7,19 +7,29 @@
 
 @Component
 public class ShowConditionParser {
+    private static final String AND_CONDITION_REGEX = "\\s\\.AND\\.\\s";
+    private static final String AND_OPERATOR = " .AND. ";
+    private static final Pattern EQUALITY_CONDITION_PATTERN = Pattern.compile("\\s*?(.*)\\s*?(=)\\s*?(\".*\")\\s*?");
 
     public ShowCondition parseShowCondition(String rawShowConditionString) throws InvalidShowConditionException {
         try {
-            Matcher matcher = Pattern.compile("\\s*?(.*)\\s*?(=)\\s*?(\".*\")\\s*?").matcher(rawShowConditionString);
-            if (matcher.find()) {
-                return new ShowCondition.Builder()
-                                .showConditionExpression(matcher.group(1).trim() + matcher.group(2).trim() + matcher.group(3).trim())
-                                .field(matcher.group(1).trim())
-                                .build();
+            String[] andConditions = rawShowConditionString.split(AND_CONDITION_REGEX);
+            ShowCondition.Builder showConditionBuilder = new ShowCondition.Builder();
+            String andOperator = "";
 
+            for (String andCondition : andConditions) {
+                Matcher matcher = EQUALITY_CONDITION_PATTERN.matcher(andCondition);
```

Comment:
can we extract this whole block inside the for loop in a private method with an explanatory name?

Reply:
cant extract block inside the for loop, but have extracted for loop out into a separate method.